### ISSUE TITLE
EZChestShop compatibility

### ISSUE THREAD

**Issue by siBalda** [GriefPrevention/GriefPrevention#2239]
### Observed Behavior

When a player clicks on a shop (a chest) in a claimed zone, the plugin sends to the player error messages because he can't open chests

### Expected Behavior

When the chest is a shop, no messages should be sent

### Reproduction steps

1- Claim a zone
2- Create a shop with a chest
3- Another player use the shop

### Server version

```shell
purpur 1.20.1
```

### GriefPrevention version

```shell
16.18.1
```

### Configuration

```yaml
not relevant
```

### Running without GriefPrevention

- [X] I attempted running the server without GriefPrevention installed.
- [X] The problem does _not_ occur when GriefPrevention is removed from the server.

### Running with only GriefPrevention

- [X] I attempted running only GriefPrevention on the server.
- [X] The issue still occurs when GriefPrevention is the only plugin running.

### Running on a fresh, clean server installation

- [X] I attempted testing for the issue on a new server.
- [X] The issue still occurs on a new server.

### Using unmodified client

- [X] I attempted testing for the issue with the vanilla client.
- [X] The issue still occurs when using the vanilla client.

### We appreciate you taking the time to fill out a bug report!

- [X] I searched for similar issues before submitting this bug report.

**Comment by siBalda** [comment]
I was wondering also if it is possible to fix that with your API but I can't find an event for interact like open a chest (I only see blockbreak event)

**Comment by Jikoo** [comment]
It is possible to modify claim permissions by listening to the [ClaimPermissionCheckEvent](https://github.com/GriefPrevention/GriefPrevention/blob/b384806e18bbd870e53323c1e8ac719d32bb33d2/src/main/java/me/ryanhamshire/GriefPrevention/events/ClaimPermissionCheckEvent.java). The triggering event should be a PlayerInteractEvent for the case you're talking about, I assume. Setting the denial reason to null allows the action.

**Comment by lerokko** [comment]
I have a similar issue with Shopkeepers. Right clicking a shop sign also sends a message that a player has no permission even though they are not trying to edit the sign, they just right click it to open the shop GUI. Should I make a new issue for that on this repository or reach out to the owner of the Shopkeepers plugin?

**Comment by RoboMWM** [comment]
> I have a similar issue with Shopkeepers. Right clicking a shop sign also sends a message that a player has no permission even though they are not trying to edit the sign, they just right click it to open the shop GUI. Should I make a new issue for that on this repository or reach out to the owner of the Shopkeepers plugin?

Yes, new issue. Signs are treated differently than chests (as far as I'm aware).

**Comment by AGuyNamedJens** [comment]
Any update to this? I have the same issue as well, i don't want players to destroy shop stands from other people while still allowing to buy or sell from the physical chests that ECS uses as shops.
